### PR TITLE
Unknown label support in `to_homogeneous`

### PR DESCRIPTION
**Pull request by puririshi98**
```
>>> x=torch_geometric.datasets.FakeHeteroDataset().data
>>> x.to_homogeneous().y
tensor([ 7.,  0.,  8.,  ..., -1., -1., -1.])
>>> x.to_homogeneous()
Data(edge_index=[2, 71136], y=[3497], node_type=[3497], edge_type=[71136])
```

### CODE REVIEW

File: torch_geometric/data/hetero_data.py [hunk 1]
```diff
@@ -891,3 +897,17 @@ def to_homogeneous_edge_index(
         edge_index = torch.cat(edge_indices, dim=-1)
 
     return edge_index, node_slices, edge_slices
+
+
+def to_homogeneous_label(data: HeteroData, ) -> Tensor:
```
Comment (Padarn) [inline torch_geometric/data/hetero_data.py:902]:
```suggestion
def to_homogeneous_label(data: HeteroData) -> Tensor:
```

File: torch_geometric/data/hetero_data.py [hunk 2]
```diff
@@ -891,3 +897,17 @@ def to_homogeneous_edge_index(
...
+    labeled_node_types = []
+    y = []
+    for i, node_type in enumerate(data.node_types):
+        if 'y' in data[node_type].keys():
+            labeled_node_types.append(i)
+            y.append(data[node_type].y)
+        else:
+            y.append(-1 * torch.ones(data[node_type].num_nodes))
+
+    if labeled_node_types:
```
Comment (Padarn) [inline torch_geometric/data/hetero_data.py:912]: when would this happen? not sure if failing silently like this makes sense

File: torch_geometric/data/hetero_data.py [hunk 3]
```diff
@@ -704,6 +705,11 @@ def _consistent_size(stores: List[BaseStorage]) -> List[str]:
         if not data.can_infer_num_nodes:
             data.num_nodes = list(node_slices.values())[-1][1]
 
+        # Combine labels
+        node_labels = to_homogeneous_label(self)
```
Comment (rusty1s) [inline torch_geometric/data/hetero_data.py:709]: Can we add a test for this?

Comment (rusty1s) [inline torch_geometric/data/hetero_data.py:709]:
```suggestion
        data.y = to_homogeneous_label(self)
```

File: torch_geometric/data/hetero_data.py [hunk 4]
```diff
@@ -891,3 +897,17 @@ def to_homogeneous_edge_index(
...
+
+def to_homogeneous_label(data: HeteroData, ) -> Tensor:
+    labeled_node_types = []
+    y = []
+    for i, node_type in enumerate(data.node_types):
+        if 'y' in data[node_type].keys():
+            labeled_node_types.append(i)
+            y.append(data[node_type].y)
+        else:
+            y.append(-1 * torch.ones(data[node_type].num_nodes))
```
Comment (rusty1s) [inline torch_geometric/data/hetero_data.py:910]: Use `torch.full` for this.

File: torch_geometric/data/hetero_data.py [hunk 5]
```diff
@@ -891,3 +897,17 @@ def to_homogeneous_edge_index(
...
+
+
+def to_homogeneous_label(data: HeteroData, ) -> Tensor:
+    labeled_node_types = []
+    y = []
+    for i, node_type in enumerate(data.node_types):
+        if 'y' in data[node_type].keys():
+            labeled_node_types.append(i)
+            y.append(data[node_type].y)
+        else:
```
Comment (rusty1s) [inline torch_geometric/data/hetero_data.py:909]: This will add node labels even in case the dataset does not have a `y` tensor at all. We should gate the for-loop by `if len(data.y_dict) > 0`.

File: torch_geometric/data/hetero_data.py (hunk 3 above)
Comment (rusty1s) [inline torch_geometric/data/hetero_data.py:709]: Only run this in self `data.y` does not exist on the homogeneous data object.

**Review by rusty1s — APPROVED** [review]
